### PR TITLE
shift unit from US$2005 to US$2017

### PR DESCRIPTION
**Pull request by LaviniaBaumstark**
## Purpose of this PR
Shift the unit of all monatary parameters and variables in REMIND to a new unit (i.e. form $US2005 to US$2017).
Most of the changes happened in the input data generation and therefore a new input data revision and minimum R-package versions are required.
Where hard coded monetary values are used in GAMS, conversion factors are introduced: s_D2015_2_D2017, sm_D2005_2_D2017, sm_D2020_2_D2017, sm_EURO2023_2_D2017

## Type of change

- [x] Refactoring

## Checklist:

- [x] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [x] I performed a self-review of my own code
- [x] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [x] All (except testing that validation data exist) automated model tests pass (`FAIL 0` in the output of `make test`)
- [x] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

## Further information (optional):

* Test runs are here: 
* Comparison of results (what changes by this PR?):

### CODE REVIEW

**Review by gabriel-abrahao — APPROVED** [review]
The MACCs conversion looks right to me, pending @fbenke-pik 's removal of the incorrect conversion in the updated `mrcommons::convertImageMacc()`.

File: modules/35_transport/edge_esm/presolve.gms [hunk 1]
```diff
@@ -13,7 +13,8 @@ if( (ord(iteration) le 25 and ord(iteration) ge cm_startIter_EDGET and (mod(ord(
     Execute "Rscript -e 'edgeTransport::iterativeEdgeTransport()'";
 
     Execute_Loadpoint 'p35_esCapCost' p35_esCapCost;
-    pm_esCapCost(t,regi,teEs_dyn35)$(t.val gt 2010 AND t.val ge cm_startyear AND t.val le 2100) = p35_esCapCost(t,regi,"%cm_GDPscen%","%cm_demScen%","%cm_EDGEtr_scen%",teEs_dyn35);
```
Comment (LaviniaBaumstark) [inline modules/35_transport/edge_esm/presolve.gms:16]: @johannah-pik could you please check, if this is correct? Thanks

File: core/declarations.gms [hunk 2]
```diff
@@ -605,9 +597,12 @@ s_MtCO2_2_GtC                                         "conversion factor from Mt
 
 s_MtCH4_2_TWa                                        "Energy content of methane. MtCH4 --> TWa: 1 MtCH4 = 1.23 * 10^6 toe * 42 GJ/toe * 10^-9 EJ/GJ * 1 TWa/31.536 EJ = 0.001638 TWa (BP statistical review)"  /0.001638/
 
-sm_h2kg_2_h2kWh                                      "convert kilogramme of hydrogen to kwh energy value." /32.5/
+s_D2015_2_D2017                                         "Convert $2015 to $2017"      /1.0292/
```
Comment (orichters) [inline core/declarations.gms:600]: Great, a new type style, as USD, US$, US, USD_ and EUR, EUR_ used in our output files were not enough 🥳

File: modules/02_welfare/ineqLognormal/datainput.gms [hunk 3]
```diff
@@ -11,23 +11,23 @@ $if %cm_less_TS% == "on"  pm_welf("2060") = 0.9;
...
+p02_inconvpen_lap(ttot,regi,"coalh2c")$(ttot.val ge 2005)     = sm_D2005_2_D2017 * 0.9;
 p02_inconvpen_lap(ttot,regi,te)$(ttot.val ge 2005) = p02_inconvpen_lap(ttot,regi,te) * 4.3 * 1E-4;            !! this is now equivalent to 1$/GJ at 1000$/per Capita in the welfare logarithm
 p02_inconvpen_lap(ttot,regi,te)$(ttot.val ge 2005) = p02_inconvpen_lap(ttot,regi,te) * (1/sm_giga_2_non) / sm_GJ_2_TWa; !! conversion util/(GJ/cap) -> util/(TWa/Gcap)
 *RP* these values are all calculated on seprod level.
 display p02_inconvpen_lap;
 $ENDIF.INCONV
 
 *BS* 2020-03-12: additional inputs for inequality
-* To Do: rename file, then also in "files" and moinput::fullREMIND.R
+*** To Do: rename file, then also in "files" and moinput::fullREMIND.R
```
Comment (orichters) [inline modules/02_welfare/ineqLognormal/datainput.gms:30]: Great you adjusted the Todo that refers to some moinput package that I have never heard of.

File: modules/21_tax/on/datainput.gms [hunk 4]
```diff
@@ -115,20 +104,7 @@ $ifThen.SEtaxRampUpParam not "%cm_SEtaxRampUpParam%" == "off"
   );
 $endif.SEtaxRampUpParam
 
-***cb 20110923 load paths for ressource export taxes
```
Comment (orichters) [inline modules/21_tax/on/datainput.gms:118]: that option was just deleted?

File: modules/32_power/IntC/datainput.gms [hunk 5]
```diff
@@ -83,19 +83,6 @@ p32_shThresholdTotVREAddIntCost(t)$(t.val > 2045) = 95;
 
 p32_FactorAddIntCostTotVRE = 1.5;
 
-$ontext
```
Comment (orichters) [inline modules/32_power/IntC/datainput.gms:86]: what was the purpose of that code, anyway?

File: modules/35_transport/edge_esm/presolve.gms (hunk 1 above)
Comment (orichters) [reply modules/35_transport/edge_esm/presolve.gms:16]: Why `t.val ge cm_startyear`? Isn't that a bit tautological?

File: modules/37_industry/subsectors/declarations.gms [hunk 6]
```diff
@@ -31,7 +31,7 @@ Parameters
   p37_mat2ue(all_enty,all_in)                                                  "Contribution of process output to ue in CES tree; Trivial if just one material per UE, as in steel [Gt/Gt]"
   p37_captureRate(all_te)                                                      "Capture rate of CCS technology"
   p37_selfCaptureRate(all_te)                                                  "Share of emissions from fossil fuels used for a CCS process which are captured by the CCS process itself"
-  p37_priceMat(all_enty)                                                       "Prices of external material input [2005$/kg = trn2005$/Gt]"
+  p37_priceMat(all_enty)                                                       "Prices of external material input [US$/kg] = [trn$US/Gt]"
```
Comment (orichters) [inline modules/37_industry/subsectors/declarations.gms:34]: Why not US$2017? I would prefer to always have the reference years in the unit.

**Review by orichters — COMMENTED** [review]
I checked the changed, added some witty comments. I cannot find any obvious reason or errors. So maybe it is in the parts that you haven't touched?

File: modules/37_industry/subsectors/declarations.gms (hunk 6 above)
Comment (LaviniaBaumstark) [reply modules/37_industry/subsectors/declarations.gms:41]: this is a pure GAMS internal parameter and the reference year depends only on the unit of the input data

**Review by dklein-pik — APPROVED** [review]

**Pull request merged**